.github/workflows/tests.yml

```diff
@@ -8,25 +8,40 @@ concurrency:
   cancel-in-progress: true
 jobs:
   jest:
-    name: Jest
+    name: 'Jest [${{ matrix.specs }}] (Node ${{ matrix.node }})'
     runs-on: ubuntu-latest
+    timeout-minutes: 10
+    strategy:
+      matrix:
+        specs: [browserify, integ, unit]
```

Comment:
slightly worried about this getting out of sync. Presumably this is an attempt to speed up the build, by parallelising more? Was the time to run all the tests prohibitive?

It's not entirely obvious that waiting for 9 GHA workers to be available is actually going to be any quicker than just waiting for 3 to be available and doing more with each worker. Given it means that you have to do all the setup actions more often, it also means you're using more GHA worker-minutes in total, which has a negative effect on all the other projects using GHA.

Just some thoughts, not a blocking comment.

Reply:
The browserify test is super basic, requires the most setup, and takes more than half the test time, it is nice to not have it block so you get a more responsive CI.

Reply:
The longer term plan is to only test one node version on every commit and then run the full test of suites when the PR enters the [Merge Queue](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/configuring-pull-request-merges/managing-a-merge-queue) - once this is available